guildwars2/characters.py

```diff
@@ -256,6 +257,273 @@ def handle_duplicates(upgrades):
             await ctx.send(first.format(user))
             await ctx.send(second)
 
+    @character.command(name="attributes")
+    @commands.cooldown(1, 10, BucketType.user)
+    async def character_attributes(self, ctx, *, character: str):
+        """Lists attributes of given character
+
+        Required permissions: characters
+        """
+        character = character.title()
+        attr_list = [
+            'defense', 'Power', 'Vitality', 'Precision', 'Toughness',
+            'Critical Chance', 'Health', 'Concentration', 'Expertise',
+            'BoonDuration', 'ConditionDamage', 'Ferocity', 'CritDamage',
+            'Healing', 'ConditionDuration', 'AgonyResistance'
+        ]
+        percentage_list = ['Critical Chance', 'CritDamage', 'ConditionDuration',
+                           'BoonDuration']
+        lvl_dict = {
+            7: [2, 10],
+            10: [11, 20],
+            14: [21, 24],
+            15: [25, 26],
+            16: [27, 30],
+            20: [31, 40],
+            24: [41, 44],
+            25: [45, 46],
+            26: [47, 50],
+            30: [51, 60],
+            34: [61, 64],
+            35: [65, 66],
+            36: [67, 70],
+            44: [71, 74],
+            45: [75, 76],
+            46: [77, 80]
+        }
+        health_group1 = {
+            28: [1, 19],
+            70: [20, 39],
+            140: [40, 59],
+            210: [60, 79],
+            280: [80, 80]
+        }
+        health_group2 = {
+            18: [1, 19],
+            45: [20, 39],
+            90: [40, 59],
+            135: [60, 79],
+            180: [80, 80]
+        }
+        health_group3 = {
+            5: [1, 19],
+            12.5: [20, 39],
+            25: [40, 59],
+            37.5: [60, 79],
+            50: [80, 80]
+        }
+
+        profession_group = {
+            "warrior": health_group1,
+            "necromancer": health_group1,
+            "revenant": health_group2,
+            "engineer": health_group2,
+            "ranger": health_group2,
+            "mesmer": health_group2,
+            "guardian": health_group3,
+            "thief": health_group3,
+            "elementalist": health_group3
+        }
+
+        ignore_list = [
+            'HelmAquatic', 'WeaponAquaticA', 'WeaponAquaticB', 'WeaponB1',
+            'WeaponB2'
+        ]
+        attr_dict = {key: 0 for (key) in attr_list}
+        runes = {}
+        await ctx.trigger_typing()
+        try:
+            results = await self.get_character(ctx, character)
+        except APINotFound:
+            return await ctx.send("Invalid character name")
+        except APIError as e:
+            return await self.error_handler(ctx, e)
+        profession = results["profession"].lower()
+        color = self.gamedata["professions"][profession]["color"]
+        icon = self.gamedata["professions"][profession]["icon"]
```

Comment:
Use get_profession() added in a recent commit

Reply:
Adapted to new get_profession()